src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt

```diff
@@ -0,0 +1,268 @@
+                    jsonDeletes++
+                }
+
+                if (jsonDeletes == 3) {
+                    deleteMetadataFromJson(org.id)
+                }
+
+                logger.info(
+                    "processing org took {} ms - org ${org.id}",
+                    timer.elapsed(TimeUnit.MILLISECONDS),
+                )
+                logger.info("progress ${index + 1}/${organizations.size}")
+                logger.info("================================")
+                logger.info("================================")
+            }
+            catch(e: Exception) {
+                logger.error("something went wrong processing org ${org.id}", e)
+            }
+        }
+
+        return true
+    }
+
+    private fun selectMetadataEntitySetIdsSql(orgId: UUID): String {
+        return """
+            SELECT
+            ${ORGANIZATION.name}->'$META_ESIDS'->>'$COLUMNS_KEY' as $COLUMNS_ESID,
+            ${ORGANIZATION.name}->'$META_ESIDS'->>'$DATASETS_KEY' as $DATASETS_ESID,
+            ${ORGANIZATION.name}->'$META_ESIDS'->>'$ORGANIZATION_KEY' as $ORGANIZATION_ESID
+            FROM ${ORGANIZATIONS.name} WHERE ${ID.name} = '$orgId'
+        """.trimIndent()
+    }
+
+    private fun getUuidFromString(orgId: UUID, entitySetId: String?): UUID? {
+        return try {
+            UUID.fromString(entitySetId)
+        } catch (e: Exception) {
+            logger.error("error converting id to uuid - org $orgId entity set id $entitySetId", e)
+            null
+        }
+    }
+
+    private fun deleteMetadataAndAuditEntitySets(orgId: UUID, entitySetId: UUID?, metadataKey :String): Boolean {
+
+        if (entitySetId == null) {
+            logger.warn("entity set id is null - org $orgId")
+            return false
+        }
+
+        try {
+
+            val auditEdgeEntitySetIds = auditService.getAuditEdgeEntitySets(AclKey(entitySetId))
+            if (auditEdgeEntitySetIds.isEmpty()) {
+                logger.warn("no audit edge entity sets - org $orgId entity set $entitySetId")
+            }
+
+            val auditRecordEntitySetIds = auditService.getAuditRecordEntitySets(AclKey(entitySetId))
+            if (auditRecordEntitySetIds.isEmpty()) {
+                logger.warn("no audit record entity sets - org $orgId entity set $entitySetId")
+            }
+
+            auditEdgeEntitySetIds.forEach { deleteEntitySet(orgId, it) }
+            auditRecordEntitySetIds.forEach { deleteEntitySet(orgId, it) }
+
+            if (deleteEntitySet(orgId, entitySetId)) {
+                return deleteMetadataFromJson(orgId, metadataKey)
+            }
+            // if the esid is all zeros AND there's no audit entity sets, it's likely the org wasn't initialized
+            // correctly. since there's no entity sets to delete, we'll try deleting metadata from json.
+            else if (
+                entitySetId.toString() == "00000000-0000-0000-0000-000000000000"
```

Comment:
UUID(0, 0) returns a UUID with all zeros

Reply:
Also, could one of `{ auditEdgeEntitySetIds, auditRecordEntitySetIds }` be empty and the other non-empty? That is, shouldn't this be `a && (b || c)`

Reply:
I specifically want both to be empty. I only want to delete from json only if there's no entity sets left to delete. If an org is in a bad state, not initialized correctly, I want to leave the json alone so I can debug later and maybe manually do the cleanup. 

Whether or not this logic is "correct" or desirable is a different question, but it is what I intended.

Reply:
Updated with `UUID(0, 0)`.